.devilbox/www/htdocs/mail.php

```diff
@@ -174,7 +174,13 @@
 								<tr id="mail-<?php echo $data['num'];?>" style="display:none">
 									<td></td>
 									<td colspan="4">
-										<?php echo $structure->body ?>
+										<?php
+										if (isset($structure->parts)) {
```

Comment:
This checks if `parts` exists, however you should also check if `parts[0]` exists:
```php
<?php
if (isset($structure->parts) && isset($structure->parts[0])) {

Reply:
True, would be nice to also know if parts exists if nothing is in there. I feel like there is more code needed to handle all cases. This was just a quick fix until I find more data

Reply:
Could it be that one part is a html version and a the second part is a text version (if it exists)?

Reply:
That was my  initial thought but they were both html. 

```
[parts] => Array
                (
                    [0] => stdClass Object
                        (
                            [headers] => Array
                                (
                                    [content-type] => text/plain; charset=UTF-8
                                    [content-transfer-encoding] => 8bit
                                )

                            [ctype_primary] => text
                            [ctype_secondary] => plain
                            [ctype_parameters] => Array
                                (
                                    [charset] => UTF-8
                                )

                            [body] => Your temporary password is: yJIG!&q



                        )

                    [1] => stdClass Object
                        (
                            [headers] => Array
                                (
                                    [content-type] => text/html; charset=UTF-8
                                    [content-transfer-encoding] => quoted-printable
                                )

                            [ctype_primary] => text
                            [ctype_secondary] => html
                            [ctype_parameters] => Array
                                (
                                    [charset] => UTF-8
                                )

                            [body] => Your temporary password is: yJIG!&q


                        )

                )
```

Reply:
```
<?php
[parts] => Array
                (
                    [0] => stdClass Object
                        (
                            [ctype_primary] => text
                            [ctype_secondary] => plain
                    [1] => stdClass Object
                            [ctype_primary] => text
                            [ctype_secondary] => html
```

Looks like the first one is plain text (where you could actually also add html, but should not be rendered) and the second one is the html version (which could also be just text, if you don't add any html stuff to it).

1. So the first one should not be rendered
2. The second one should be rendered.

What do you think.

Reply:
I found this 
https://stackoverflow.com/questions/7166922/extracting-the-body-of-an-email-from-mbox-file-decoding-it-to-plain-text-regard

According to the link above there might be subparts to parts. I tried to find a php library that might do the work for us but I could not find one.

I believe we should read the content type and go from there
[content-type] => multipart/alternative; boundary="B_ALT_5a28f33eda0b0"
[content-type] => text/plain; charset="UTF-8"

Let me know if you find any other resources about this.


or we can keep it simple like:
```
<?php if (isset($structure->body)): ?>
    <?php echo $structure->body ?>
<?php elseif(isset($structure->parts[1]->body)): ?>
    <?php echo isset($structure->parts[1]->body) ?>
<?php elseif(isset($structure->parts[0]->body)): ?>
    <?php echo isset($structure->parts[0]->body) ?>
<?php else: ?>
    <?php echo '<div class="alert alert-warning" role="alert">
        No valid body found
    </div>' ?>
<?php endif; ?>
```

Reply:
I think this will be sufficient for the current use-case. I would additionally surround the `text/plain`-based email with `htmlentities(..)`, so its made sure that it is not rendered, as it is actually text-only

In case we hit some edges, it can still be further improved later upon that.